R/copy-model-from.R

```diff
@@ -9,9 +9,11 @@
 #' ancestry. See ["Using based_on field"
 #' vignette](../articles/using-based-on.html) for details.
 #' @param .parent_mod Model to copy from
-#' @param .new_model Path to write new model files to **without file extension**.
-#'   Function will create both `{.new_model}.yaml` and a new model file based on
-#'   this path.
+#' @param .new_model Path, relative to location of `.parent_model`, to write new
+#'   model files to. Should have **no file extension** because function will
```

Comment:
We want to allow the new model to have a period in its absolute model path, right? There isn't actually any logic to enforce that `.new_model` does not have a file extension, so the issue will arise in `copy_nonmem_model_from()` when `fs::path_set_ext()` is called. That call will _replace_ the existing "extension," so that `copy_control_stream()` will not copy the model file to the correct path, and then `find_nonmem_model_file_path()` will not find it.

```r
library(rbabylon)
parent <- read_model("~/git/rbabylon/tests/testthat/model-examples/1.yaml")
copy_model_from(parent, "foo.bar", "baz")
#> Error in find_nonmem_model_file_path(res[[ABS_MOD_PATH]], .check_exists = TRUE): No model file found at /data/home/seanw/git/rbabylon/tests/testthat/model-examples/foo.bar.ctl or /data/home/seanw/git/rbabylon/tests/testthat/model-examples/foo.bar.mod. Please put relevant model file in that location.
```

Reply:
mmm, interesting. I saw your comment about this somewhere else I must've forgotten to deal with it. I'll add a test for that too.

Reply:
We might want to remove the "no file extension" language. It may be clearer to have something like

> Path to the new model, either absolute or relative to the path to `.parent_model`. Represents an absolute model path, which is the path to the YAML file and model file, both without extension, and (possibly) the output directory. Numeric values will be coerced to character. See examples for usage.

and then `@examples` showing the allowed options.

```r
#' @examples
#' \dontrun{
#' parent <- read_model("/foo/parent")
#'
#' # create model file at /bar/child.ctl and YAML at /bar/child.yaml
#' copy_model_from(parent, "/bar/child", "child model with absolute path")
#'
#' # create model file at /foo/child.ctl and YAML at /foo/child.yaml
#' copy_model_from(parent, "child", "relative to parent model path")
#'
#' mod1 <- read_model("/path/to/1")
#'
#' # create model file at /path/to/2.ctl and YAML at /path/to/2.yaml
#' copy_model_from(mod1, 2, "numeric input works")
#'
#' # create model file at /path/to/100.1.ctl and YAML at /path/to/100.1.yaml
#' copy_model_from(mod1, 100.1, "a period is okay")
#' }
```